app/workers/share_analytics_updater.rb

```diff
@@ -17,16 +24,27 @@ def update_collection
     update_shares
   end
 
+  def update_share(id)
+    share = Share::Button.find id
+
+
+    if share and share.sp_id.present?
+      response = Net::HTTP.post_form(@uri, {key: @sp_api_key, id: share.sp_id})
+      raise_or_update(response, share)
+    end
+  end
+
   private
 
   def update_shares
     @collection.each do |button|
-      begin
-        request_analytics_from_sp(button)
-      rescue ::ShareProgressApiError
-        puts $!.message
-        next
-      end
+      Aws::SQS::Client.new.send_message({
+        queue_url:    ENV['SQS_ENDPOINT'],
+        message_body: {
+          type: 'update_share',
+          share_id: button.id
+        }
+      }.to_json)
```

Comment:
@Tuuleh with this the cron would spit out messages into the queue, so creating one job for each share.